deploy/ansible/roles/base/tasks/main.yml

```diff
@@ -2,17 +2,44 @@
 # or the project is cloned or built. All system dependencies should be
 # installed here.
 
+- name: ensure apt cache is updated
+  apt: update_cache=yes cache_valid_time=3600
+
+- name: ensure all packages are upgraded safely
+  apt: upgrade=safe
+  when: env != "development"
+
 - name: add keys to apt
   apt_key: url={{item}} state=present
   with_items: apt_keys
 
 - name: add ppas to apt
-  apt_repository: repo="{{item}}" update_cache=yes
+  apt_repository:
+    repo: "{{item}}"
+    state: present
   with_items: apt_ppas
 
 - name: install apt packages
-  apt: name={{item}} state=latest update_cache=yes
+  apt:
+    name: "{{item}}"
+    state: latest
   with_items: apt_packages
 
 - name: update npm to latest
   npm: name=npm state=latest global=yes
+
+- name: install global npm modules
+  npm:
+    name={{item.name}}
+    state=present
+    version={{ item.version | default(omit) }}
+    global=yes
+  with_items: npm_global_modules
+
+- name: install ruby gems
+  gem:
+    name={{item.name}}
+    state=present
+    version={{ item.version | default(omit) }}
+    user_install=no
+  with_items: ruby_gems
```

Comment:
I originally added this without the `user_install` line. Looks like adding that line didn't change the result of the `present` test, so it kept passing as "OK" instead of updating the install. Destroyed and recreated the box, and it looks like it installed.